src/metabase/cmd.clj

```diff
@@ -52,8 +52,9 @@
   (let [options        {:keep-existing? (boolean (some #{"--keep-existing"} opts))
                         :dump-plaintext? (boolean (some #{"--dump-plaintext"} opts)) }
         return-code    ((resolve 'metabase.cmd.dump-to-h2/dump-to-h2!) h2-filename options)]
-    (when (pos-int? return-code)
-      (system-exit! return-code))))
+    (if (pos-int? return-code)
```

Comment:
I'm trying to keep the same interface of "if return positive int" because it was already there, but, it feels to me this case won't be reached every, because `dump-to-h2!` either ends ok, or throws an exception (copy/copy! throws them).

Am I missing something? if the pos-int? interface is not used at all and we rely on the exception being thrown or not, I can simplify this a bit more

Reply:
I think that's the issue - the exception being thrown leads down a path where there is no `(system-exit! 1)` or anything. I think the main command dispatcher may need a try...catch to handle exceptional conditions